### ISSUE TITLE
[java] StringToString rule false-positive with field access

### ISSUE THREAD

**Issue by niktekusho** [pmd/pmd#2080]
**Affects PMD Version:** `6.18.0`

**Rule:** [StringToString](https://pmd.github.io/latest/pmd_rules_java_performance.html#stringtostring)

**Description:**

*Attached code should help to understand better the following description.*

We found a false-positive case for the rule [String.toString()](https://pmd.github.io/pmd-6.18.0/pmd_rules_java_performance.html#stringtostring). 
When you assign an object field (of type `String`) to a field of another object (not of type `String`) calling `toString()`, the rule emits a warning about calling toString on a String object *only when the field names match* (`value = otherObj.value.toString()`). If either one of the field name changes, Pmd stops emitting the warning (workaround).

Since the enum is not a `String` object, we think that the warning is a false-positive triggered by the field name equality. 

**Code Sample demonstrating the issue:**

*See attached project for a ready-to-go reproduction case.*

``` java
public enum AnEnum {

	A,
	B,
	C;

}

public class AnEnumUser {

	private final String value;

	public AnEnumUser(final OtherEnumUser otherUser) {
                // The following line triggers the "false-positive" warning
		value = otherUser.value.toString();
	}

	public String getValue() {
		return value;
	}

}

public class OtherEnumUser {
	public final AnEnum value;

	public OtherEnumUser(final AnEnum value) {
		this.value = value;
	}
}

public final class Main {
	private Main() {
	}

	public static void main(final String[] args) {
		final OtherEnumUser a = new OtherEnumUser(AnEnum.A);
		System.out.println("Original " + a.value);

		final AnEnumUser aCopy = new AnEnumUser(a);
		System.out.println("Copy " + aCopy.getValue());

	}
}
```

**Running PMD through:** *Gradle*

[Attached reproduction project (forgive the eclipse files)](https://github.com/pmd/pmd/files/3761945/pmd-string-to-string-false-positive.zip)

**Comment by adangel** [comment]
Shorter reproduction sample:

```xml
    <test-code>
        <description>[java] StringToString rule false-positive with field access #2080</description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
public final class Issue2080 {
    String value;

    public void foo() {
        A a = new A();
        value = a.value.toString(); // false positive
    }

    class A {
        B value;
    }

    class B { }
}
]]></code>
    </test-code>
```
